KalturaWowzaServer/src/main/java/com/kaltura/media_server/services/DiagnosticsProvider.java

```diff
@@ -78,6 +88,29 @@ public void execute(HashMap<String,Object> data, HashMap<String,String > quarySt
             data.put("dateStarted", dateStarted);
             data.put("timeRunning", timeRunning);
             data.put("hostName", hostName);
+            data.put("CpuUsage", CpuUsage);
+
+            HashMap<String,Object> entriesData = getStreamsData(appInstance);
+            data.putAll(entriesData);
+        }
+
+        public HashMap<String,Object> getStreamsData(IApplicationInstance appInstance) {
+            HashMap<String,Object> entriesData = new HashMap<String,Object>();
+            int in = 0, out = 0;
+            for (IMediaStream stream : appInstance.getStreams().getStreams()) {
+                if (!( stream instanceof MediaStreamLive))
+                    continue;
+                if (stream.isTranscodeResult())
+                    out++;
+                else
+                    in++;
+            }
+            entriesData.put("Input Streams", in);
+            entriesData.put("OutPut Streams", out);
+
+            Set<String> playingEntriesList = Utils.getEntriesFromApplication(appInstance);
```

Comment:
you already do a pass over all streams above, it's double the work... (+ calling appInstance.getStreams() twice)
i think it's better to just copy the line that count the entries